polaris-service/src/main/java/org/apache/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -1874,10 +1874,21 @@ private boolean sendNotificationForTableLike(
                         .getMetaStoreManager()
                         .generateNewEntityId(getCurrentPolarisContext())
                         .getId())
+                .setLastNotificationTimestamp(request.getPayload().getTimestamp())
                 .build();
       } else {
+        // If the notification timestamp is outdated, we should not update the table
+        if (entity.getLastNotificationTimestamp() != null
```

Comment:
Instead of `null`, can we use `Optional`?

Reply:
Done!